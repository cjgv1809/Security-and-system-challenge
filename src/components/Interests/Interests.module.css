.interests {
  padding: 1.5rem;
  background: var(--color-background-light);
  border-radius: 16px;
  box-shadow: 0 4px 30px var(--color-box-shadow);
  backdrop-filter: blur(6px);
  -webkit-backdrop-filter: blur(6px);
  border: 1px solid var(--color-border-light);
  width: min(100%, 600px);
  margin: 0 auto;
}

.title {
  font-size: 1.5rem;
  margin-bottom: 0.625rem;
}

.list {
  list-style-type: none;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  gap: 0.625rem;
}

.item {
  font-size: 1rem;
  color: var(--color-item-text);
  padding: 0.3125rem 0.625rem;
  background-color: var(--color-item-background);
  border-radius: 15px;
}

.noItemText {
  font-size: 1rem;
  color: var(--color-text-light);
}
