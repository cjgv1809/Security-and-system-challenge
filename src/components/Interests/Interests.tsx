import React from "react";
import styles from "./Interests.module.css";

type InterestsProps = {
  interests: string[];
};

const Interests: React.FC<InterestsProps> = ({ interests }) => {
  return (
    <section className={styles.interests}>
      <h2 className={styles.title}>Intereses</h2>
      <ul className={styles.list}>
        {interests.map((interest, index) => (
          <li key={index} className={styles.item}>
            {interest}
          </li>
        ))}
      </ul>
    </section>
  );
};

export default Interests;
