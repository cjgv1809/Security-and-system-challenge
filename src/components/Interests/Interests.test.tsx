import React from "react";
import { render, screen } from "@testing-library/react";
import Interests from "./Interests";

describe("Interests", () => {
  it("renders the list of interests", () => {
    const interests = ["Coding", "Reading", "Traveling"];
    render(<Interests interests={interests} />);

    expect(screen.getByText("Intereses")).toBeInTheDocument();
    interests.forEach((interest) => {
      expect(screen.getByText(interest)).toBeInTheDocument();
    });
  });

  it("renders a message if there are no interests", () => {
    render(<Interests interests={[]} />);

    expect(screen.getByText("No hay intereses")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const interests = ["Coding", "Reading", "Traveling"];
    const { container } = render(<Interests interests={interests} />);

    expect(container).toMatchSnapshot();
  });
});
