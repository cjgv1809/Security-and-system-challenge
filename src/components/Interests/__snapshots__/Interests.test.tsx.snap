// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interests matches snapshot 1`] = `
<div>
  <section
    class="interests"
  >
    <h2
      class="title"
    >
      Intereses
    </h2>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        Coding
      </li>
      <li
        class="item"
      >
        Reading
      </li>
      <li
        class="item"
      >
        Traveling
      </li>
    </ul>
  </section>
</div>
`;
